cylc/flow/scripts/install.py

```diff
@@ -32,25 +29,35 @@
 A .service directory will also be created and used for server authentication
 files at run time.
 
+If the argument REG is used, Cylc will search for the workflow in the list of
+directories given by `global.cylc[install]source dirs`, and install the first
+match. Otherwise, the workflow in the current working directory, or the one
+specified by --directory, will be installed.
 
 Workflow names can be hierarchical, corresponding to the path under ~/cylc-run.
 
 Examples:
-  # Install workflow dogs/fido from $PWD
-  # (with run directory ~/cylc-run/dogs/fido/run1)
-  # (if "run1" exists this will increment)
+  # Install workflow "dogs/fido" from the first match in
+  # `global.cylc[install]source dirs`, e.g. ~/cylc-src/dogs/fido/flow.cylc
+  # (with run directory ~/cylc-run/dogs/fido/run1 (if "run1" already exists,
+  # this will increment))
   $ cylc install dogs/fido
 
-  # Install $PWD/flow.cylc with specified flow name: fido
+  # Install $PWD/flow.cylc as "rabbit", if $PWD is ~/bunny/rabbit
+  # (with run directory ~/cylc-run/rabbit/run1)
+  $ cylc install
+
+  # Install $PWD/flow.cylc as "fido", regardless of what $PWD is
   # (with run directory ~/cylc-run/fido/run1)
   $ cylc install --flow-name=fido
 
-  # Install PATH/TO/FLOW/flow.cylc
-  $ cylc install --directory=PATH/TO/FLOW
+  # Install $PWD/bunny/rabbit/flow.cylc as "bunny/rabbit"
+  # (with run directory ~/cylc-run/bunny/rabbit/run1)
+  $ cylc install --directory=bunny/rabbit
 
-  # Install cats/flow.cylc
+  # Install $PWD/cats/flow.cylc as "cats", overriding the run1, run2, run3 etc
+  # structure.
```

Comment:
I'm having problems getting these examples to work. I shall wait until this PR is marked ready before reviewing.

Reply:
Should be the same use cases that are covered by the cylc-install functional tests, passing for me and on GH Actions